src/lib/InstantSearch.svelte

```diff
@@ -36,7 +36,12 @@
 
     search.start();
 
-    searchClient.addAlgoliaAgent("svelte-algolia-instantsearch", "1.0.0");
+    // Try to add a custom user agent, but don't crash on fail since some instantsearch.js adapters don't have this.
+    try {
+        searchClient.addAlgoliaAgent("svelte-algolia-instantsearch", "1.0.0");
+    } catch (e) {
+        // ignore
+    }
```

Comment:
Hey, thanks for the PR !

```suggestion
    if('addAlgoliaAgent' in searchClient) {
      searchClient.addAlgoliaAgent("svelte-algolia-instantsearch", "1.0.0");
    }
```

I would prefer this, it's shorter and wouldn't dismiss eventual exceptions thrown by an actual `addAlgoliaAgent` function

Reply:
I like this much better as well, thanks for the input!